rasa/nlu/components.py

```diff
@@ -203,6 +211,69 @@ def validate_required_components_from_data(
             "no response selector component specified inside NLU pipeline."
         )
 
+    if data.entity_examples and not any_components_in_pipeline(
+        TRAINABLE_EXTRACTORS, pipeline
+    ):
+        raise_warning(
+            "You have defined training data consisting of entity examples, but "
+            "your NLU pipeline does not include an entity extractor trained on "
+            "your training data. To extract entity examples, add one of "
+            f"{TRAINABLE_EXTRACTORS} to your pipeline."
+        )
```

Comment:
I guess DIET is a trainable extractor now? Kind of? 🤔

Reply:
yes

Reply:
Actually, if we want to be really correct, we should check in case of DIET if the option `entity_extraction` is set to `True` otherwise it will not predict any entities.